### PR TITLE
refactor(proofs)!: rename ProofRecord to ProofExchangeRecord

### PR DESCRIPTION
**Pull request by NB-MikeRichardson**

### CODE REVIEW

File: packages/core/src/storage/migration/__tests__/__snapshots__/0.2.test.ts.snap [hunk 1]
```diff
@@ -165,7 +165,7 @@ Object {
       "state": "done",
       "threadId": "7fcfc074-43ac-43cc-92b9-76afceeebe82",
     },
-    "type": "ProofRecord",
+    "type": "ProofExchangeRecord",
```
Comment (TimoGlastra) [inline packages/core/src/storage/migration/__tests__/__snapshots__/0.2.test.ts.snap:168]: We should keep the type as 'ProofRecord' in the ProofExchangeRecord

File: packages/core/src/modules/proofs/repository/ProofExchangeRecord.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export type DefaultProofTags = {
 
 // T-TODO: rename to proof exchange record
```
Comment (TimoGlastra) [inline packages/core/src/modules/proofs/repository/ProofExchangeRecord.ts:31]: Can you remove this comment?

**Review by TimoGlastra — COMMENTED** [review]
Nice! Some small suggestions but otherwise LGTM!

**Comment by genaris** [comment]
This refactoring would represent a breaking change, right? In that case I'll note it for the migration guide. Would it be merged as _refactor(proofs)!: Rename ProofRecord to ProofExchangeRecord_?

**Comment by TimoGlastra** [comment]
@NB-MikeRichardson CI is failing it seemes

**Comment by NB-MikeRichardson** [comment]
> @NB-MikeRichardson CI is failing it seems. I think you need to regenerate the snapshot files

I'll take a look

**Review by TimoGlastra — APPROVED** [review]

**Pull request merged**